packages/metrics-handler/index.js

```diff
@@ -0,0 +1,57 @@
+function responseHelper(body) {
+  return {
+    statusCode: 200,
+    body: JSON.stringify(body),
+  };
+}
+
+const STARTER_KITS_JSON_URL = 'https://raw.githubusercontent.com/thisdot/starter.dev/main/starter-kits.json';
+const GOOGLE_MEASUREMENT_PROTOCOL_ENDPOINT = (measurement_id, api_secret) => `https://www.google-analytics.com/debug/mp/collect?measurement_id=${measurement_id}&api_secret=${api_secret}`;
+
+module.exports.handler = async (event) => {
+  const MEASUREMENT_ID = process.env.GOOGLE_ANALYTICS_MEASUREMENT_ID;
+  const API_SECRET = process.env.GOOGLE_ANALYTICS_API_SECRET;
+  const selectedStarterKit = JSON.parse(event.body || null)?.starterKit;
+
+  if (!selectedStarterKit) {
+    console.log('No kit found');
+    return responseHelper('No kit found');
+  }
+  const starterKitsRequest = await fetch(STARTER_KITS_JSON_URL);
+  if (!starterKitsRequest.ok) {
+    console.log('Could not fetch starter kits for validation, tracking aborted');
+    return responseHelper('Could not fetch starter kits for validation, tracking aborted')
+  }
+
+  const starterKitsJSON = await starterKitsRequest.json();
+  const starterKits = new Set()
+  if (typeof starterKitsJSON === 'object' && starterKitsJSON !== null) {
+    Object.keys(starterKitsJSON).forEach((kit) => starterKits.add(kit))
+  }
```

Comment:
There is no need to iterate manually. Set constructor accepts iterable as a parameter
```suggestion
  const starterKits = new Set(
    typeof starterKitsJSON === 'object' && starterKitsJSON !== null
      ? Object.keys(starterKitsJSON)
      : []
  )
```